packages/core/tests/helpers.ts

```diff
@@ -338,6 +376,49 @@ export async function issueCredential({
   }
 }
 
+export async function issueConnectionLessCredential({
+  issuerAgent,
+  holderAgent,
+  credentialTemplate,
+}: {
+  issuerAgent: Agent
+  holderAgent: Agent
+  credentialTemplate: Omit<CredentialOfferTemplate, 'autoAcceptCredential'>
+}) {
+  // eslint-disable-next-line prefer-const
+  let { credentialRecord: issuerCredentialRecord, offerMessage } = await issuerAgent.credentials.createOutOfBandOffer({
+    ...credentialTemplate,
+    autoAcceptCredential: AutoAcceptCredential.ContentApproved,
+  })
+
+  const credentialRecordPromise = waitForCredentialRecord(holderAgent, {
+    threadId: issuerCredentialRecord.threadId,
+    state: CredentialState.OfferReceived,
+  })
```

Comment:
Is there any benefit of making a Promise variable rather than `awat`ing it later at line 400?

Reply:
I'm in the progress of refactoring the method to work with observables. We need to start listening before calling the method that will emit the event. Otherwise we start listening potentially after the event has already been emitted. So we need to start the listener, make the function call and then await. 

I've partly refactored some other call to use ReplaySubject so we can replay all events that came in.